lib/dialects/mysql2/index.js

```diff
@@ -1,35 +1,31 @@
 // MySQL2 Client
 // -------
-const { inherits } = require('util');
 const Client_MySQL = require('../mysql');
 const Transaction = require('./transaction');
 
 // Always initialize with the "QueryBuilder" and "QueryCompiler"
 // objects, which extend the base 'lib/query/builder' and
 // 'lib/query/compiler', respectively.
-function Client_MySQL2(config) {
-  Client_MySQL.call(this, config);
-}
-inherits(Client_MySQL2, Client_MySQL);
-
-Object.assign(Client_MySQL2.prototype, {
-  // The "dialect", for reference elsewhere.
-  driverName: 'mysql2',
-
+class Client_MySQL2 extends Client_MySQL {
   transaction() {
     return new Transaction(this, ...arguments);
-  },
+  }
 
   _driver() {
     return require('mysql2');
-  },
-
+  }
   validateConnection(connection) {
     if (connection._fatalError) {
       return false;
     }
     return true;
-  },
+  }
+}
+
+// until node 12+
+Object.assign(Client_MySQL2.prototype, {
+  // The "dialect", for reference elsewhere.
+  driverName: 'mysql2',
```

Comment:
This can't be simply a getter?

Reply:
Users could override them using assign in constructor, declaring as getter could break their logic